enroll/certificates.go

```diff
@@ -1,22 +1,39 @@
 package enroll
 
 import (
+	"crypto/x509"
+	"encoding/pem"
 	"errors"
 	"golang.org/x/crypto/pkcs12"
 	"io/ioutil"
 )
 
-const PushTopicASN1 string = "0.9.2342.19200300.100.1.1"
+const (
+	certificatePEMBlockType string = "CERTIFICATE"
+	PushTopicASN1                  = "0.9.2342.19200300.100.1.1"
+)
 
-func GetPushTopicFromPKCS12(certPath string, certPass string) (string, error) {
+func GetPushTopicFromCert(certPath, certPass, keyPath string) (string, error) {
 	certData, err := ioutil.ReadFile(certPath)
 	if err != nil {
 		return "", err
 	}
 
-	_, cert, err := pkcs12.Decode(certData, certPass)
-	if err != nil {
-		return "", err
+	var cert *x509.Certificate
```

Comment:
Let's talk structure/architecture at some point. I'd like to see the capability for multiple MDM "servers" (that is push topics+URLs) supported under one single daemon/server.